test/cljs/harja/testutils.cljs

```diff
@@ -63,3 +63,40 @@
    :after #(do (clear-timeouts)
                (reset! k/testmode nil)
                (reset! fake-palvelukutsut {}))})
+
+(defn =marginaalissa?
+  "Palauttaa ovatko kaksi lukua samat virhemarginaalin sisällä. Voi käyttää esim. doublelaskennan
+  tulosten vertailussa. Oletusmarginaali on 0.05"
+  ([eka toka] (=marginaalissa? eka toka 0.05))
+  ([eka toka marginaali]
+   (< (Math/abs (double (- eka toka))) marginaali)))
+
+(defn tarkista-map-arvot
+  "Tarkistaa, että mäpissä on oikeat arvot. Numeroita vertaillaan =marginaalissa? avulla, muita
+  = avulla. Tarkistaa myös, että kaikki arvot ovat olemassa. Odotetussa mäpissa saa olla
+  ylimääräisiä avaimia."
+  [odotetut saadut]
+  (doseq [k (keys odotetut)
+          :let [odotettu-arvo (get odotetut k)
+                saatu-arvo (get saadut k ::ei-olemassa)]]
+    (cond
+      (= saatu-arvo ::ei-olemassa)
+      (is false (str "Odotetussa mäpissä ei arvoa avaimelle: " k
+                     ", odotettiin arvoa: " odotettu-arvo))
+
+      (and (number? odotettu-arvo) (number? saatu-arvo))
+      (is (=marginaalissa? odotettu-arvo saatu-arvo)
+          (str "Saatu arvo numeroavaimelle " k " ei marginaalissa, odotettu: "
+               odotettu-arvo ", saatu: "
+               saatu-arvo))
+
+      ;; (instance? java.util.Date odotettu-arvo)
```

Comment:
Tässä varmaa pitää käyttää `(instance? goog.date/DateTime odotettu-arvo)`, ku ollaan cljs puolella. harja.pvm ns:ssa käytetään ainaki `DateTime`:ä js:n `Date`:n sijasta, mutta en sitten tiedä käytetäänkö jossain päin `js/Date`:a.